Lesson22

Pavel3879 opened the pull request:
Уровень №2
Практика №2

Pavel3879 committed:
Практика №3

Pavel3879 committed:
убрал вывод загаданного слова

Pavel3879 committed:
Практика №4

Pavel3879 committed:
Практика №4

Pavel3879 committed:
Практика №5

Pavel3879 committed:
Практика №6

Pavel3879 committed:
Практика №7

Pavel3879 committed:
Практика №8

Pavel3879 committed:
Уровень №3
Практика №1

Pavel3879 committed:
Уровень №3
Практика №2

Pavel3879 committed:
Уровень №3
Практика №3

Pavel3879 committed:
Занятие 4
Практика 4

Pavel3879 committed:
Уровень №3
Практики 5 и 6

Pavel3879 committed:
Уровень №3
Практика №7

Pavel3879 committed:
почистил

Pavel3879 committed:
Уровень №2
Практика №1

Pavel3879 committed:
Уровень №2
Практика №2

Ivan-Igorevich commented on Practical/src/lesson22/MainClass.java:
```diff
@@ -0,0 +1,53 @@
+
+        for (int i = 0; i < strings.length; i++) {
+            for (int j = 0; j < strings[i].length; j++) {
+                v = Integer.parseInt(strings[i][j]);
```
я бы по факту возникновения NFE бросал какое-нибудь своё собственное исключение, например

Ivan-Igorevich commented on Practical/src/lesson22/MainClass.java:
```diff
@@ -0,0 +1,53 @@
+        for (int i = 0; i < strings1.length; i++) {
+            strings2 = strings1[i].split(" ");
+            checkArrayLength(strings2, LENGTH);
+            strings[i] = strings2.clone();
```
а зачем клон?

Pavel3879 commented:
Бросить свое исключение — думал, но что-то меня отговорило…
С колном перегнул, с массивом  strings2 никаких манипуляций больше не произвожу, можно простым присвоением было обойтись.
  
>Вторник, 5 ноября 2019, 15:50 +03:00 от Ivan I. Ovchinnikov <notifications@github.com>:
> 
>@Ivan-Igorevich commented on this pull request.
>----------------------------------------------------------------------
>In  Practical/src/lesson22/MainClass.java :
>> +        } catch (NumberFormatException e) {
>
>+            System.out.println(e.getMessage());
>
>+        } catch (IllegalArgumentException e) {
>
>+            System.out.println(e.getMessage());
>
>+        }
>
>+
>
>+
>
>+    }
>
>+
>
>+    static float calc(String[][] strings) throws NumberFormatException {
>
>+        int v;
>
>+        float result = 0;
>
>+
>
>+        for (int i = 0; i < strings.length; i++) {
>
>+            for (int j = 0; j < strings[i].length; j++) {
>
>+                v = Integer.parseInt(strings[i][j]);
>
>
>я бы по факту возникновения NFE бросал какое-нибудь своё собственное исключение, например
>----------------------------------------------------------------------
>In  Practical/src/lesson22/MainClass.java :
>> +                v = Integer.parseInt(strings[i][j]);
>
>+                result += v;
>
>+            }
>
>+        }
>
>+        return result / 2;
>
>+    }
>
>+
>
>+    static void stringToArray2(String s, String[][] strings) throws IndexOutOfBoundsException {
>
>+        String[] strings1;
>
>+        String[] strings2;
>
>+        strings1 = s.split("\n");
>
>+        checkArrayLength(strings1, LENGTH);
>
>+        for (int i = 0; i < strings1.length; i++) {
>
>+            strings2 = strings1[i].split(" ");
>
>+            checkArrayLength(strings2, LENGTH);
>
>+            strings[i] = strings2.clone();
>
>
>а зачем клон?
>—
>You are receiving this because you authored the thread.
>Reply to this email directly,  view it on GitHub , or  unsubscribe .
 
 
--
Павел Вытягов

 

Pavel3879 committed:
исправления
